### PR TITLE
Refactor path and inode arguments related code

### PR DESCRIPTION
**Pull request by cosmo0920**
**Which issue(s) this PR fixes**: 
Follows up #3182 

**What this PR does / why we need it**: 
Unify path and inode arguments.

**Docs Changes**:
No needed.

**Release Note**: 
Same as title.

### CODE REVIEW

File: lib/fluent/plugin/in_tail.rb [hunk 1]
```diff
@@ -386,78 +386,75 @@ def setup_watcher(path_with_inode, pe)
...
 
     # refresh_watchers calls @tails.keys so we don't use stop_watcher -> start_watcher sequence for safety.
-    def update_watcher(path_with_inode, pe)
+    def update_watcher(target_info, pe)
       log.info("detected rotation of #{path}; waiting #{@rotate_wait} seconds")
 
       if @pf
-        unless pe.read_inode == @pf[path_with_inode.path, pe.read_inode].read_inode
+        target_info_from_position_entry = TargetInfo.new(target_info.path, pe.read_inode)
+        unless pe.read_inode == @pf[target_info_from_position_entry].read_inode
```
Comment (repeatedly) [inline lib/fluent/plugin/in_tail.rb:447]: Call `pe.read_inode` twice. Use local variable is better like `pe_inode = pe.read_inode`

File: lib/fluent/plugin/in_tail.rb [hunk 2]
```diff
@@ -386,78 +386,75 @@ def setup_watcher(path_with_inode, pe)
...
           log.debug "Skip update_watcher because watcher has been already updated by other inotify event"
           return
         end
       end
 
-      target_info = TargetInfo.new(path_with_inode.path, pe.read_inode)
       rotated_tw = @tails[target_info]
-
-      new_target_info = TargetInfo.new(path_with_inode.path, path_with_inode.ino)
+      new_target_info = TargetInfo.new(target_info.path, target_info.ino)
```
Comment (repeatedly) [inline lib/fluent/plugin/in_tail.rb:454]: `target_info.dup` has a problem?

File: lib/fluent/plugin/in_tail/position_file.rb [hunk 3]
```diff
@@ -37,32 +37,32 @@ def initialize(file, follow_inodes, existing_paths, logger: nil)
...
+            @map[target_info.path] = FilePositionEntry.new(@file, @file_mutex, seek, 0, 0)
           end
         }
       end
 
-      def unwatch(path, inode)
+      def unwatch(target_info)
         if @follow_inodes
-          if (entry = @map.delete(inode))
+          if (entry = @map.delete(target_info.ino))
```
Comment (repeatedly) [inline lib/fluent/plugin/in_tail/position_file.rb:61]: How about `@map.delete(@follow_inodes ? target_info.ino : target_info.path)`?

File: lib/fluent/plugin/in_tail.rb (hunk 1 above)
Comment (cosmo0920) [reply lib/fluent/plugin/in_tail.rb]: Fixed in https://github.com/fluent/fluentd/pull/3196/commits/e4abf5847f2f0a29dc71319f3be1fb22b989ee9b.

File: lib/fluent/plugin/in_tail.rb (hunk 2 above)
Comment (cosmo0920) [reply lib/fluent/plugin/in_tail.rb]: No issue.
Fixed in https://github.com/fluent/fluentd/pull/3196/commits/828f91496763e8069a6e3a1c082a1f8d0b492d04.

File: lib/fluent/plugin/in_tail/position_file.rb (hunk 3 above)
Comment (cosmo0920) [reply lib/fluent/plugin/in_tail/position_file.rb]: Fixed in https://github.com/fluent/fluentd/pull/3196/commits/bc0602657817f3e20d34fe5678ff1650bfc153d8.

File: lib/fluent/plugin/in_tail.rb [hunk 4]
```diff
@@ -386,78 +386,76 @@ def setup_watcher(path_with_inode, pe)
...
-        unless pe.read_inode == @pf[path_with_inode.path, pe.read_inode].read_inode
+        pe_inode = pe.read_inode
+        target_info_from_position_entry = TargetInfo.new(target_info.path, pe_inode)
+        unless pe_inode == @pf[target_info_from_position_entry].read_inode
           log.debug "Skip update_watcher because watcher has been already updated by other inotify event"
           return
         end
       end
 
-      target_info = TargetInfo.new(path_with_inode.path, pe.read_inode)
```
Comment (repeatedly) [inline lib/fluent/plugin/in_tail.rb:454]: The inode value is different from `target_info.ino`.
Removing this line is safe?

Comment (cosmo0920) [reply lib/fluent/plugin/in_tail.rb:454]: Oh. I've overlooked it.

Comment (cosmo0920) [reply lib/fluent/plugin/in_tail.rb:454]: Fixed in https://github.com/fluent/fluentd/pull/3196/commits/a106a0317111e50f705fd861dcaf51d7da03acd7.

**Review by repeatedly — APPROVED** [review]

**Pull request merged**